python_modules/libraries/dagster-aws/dagster_aws/pipes.py

```diff
@@ -241,3 +267,144 @@ def run(
+    """
+
+    def __init__(
+        self,
+        client: boto3.client,
+        context_injector: PipesContextInjector,
+        message_reader: Optional[PipesMessageReader] = None,
+    ):
+        self._client = client
+        self._context_injector = context_injector
+        self._message_reader = message_reader or PipesCloudWatchMessageReader()
+
+    @classmethod
+    def _is_dagster_maintained(cls) -> bool:
+        return True
+
+    def run(
+        self,
+        *,
+        job_name: str,
+        context: OpExecutionContext,
+        extras: Optional[Dict[str, Any]] = None,
+        arguments: Optional[Mapping[str, Any]] = None,
+        job_run_id: Optional[str] = None,
+        allocated_capacity: Optional[int] = None,
+        timeout: Optional[int] = None,
+        max_capacity: Optional[float] = None,
+        security_configuration: Optional[str] = None,
+        notification_property: Optional[Mapping[str, Any]] = None,
+        worker_type: Optional[str] = None,
+        number_of_workers: Optional[int] = None,
+        execution_class: Optional[Literal["FLEX", "STANDARD"]] = None,
+    ):
+        """Start a Glue job, enriched with the pipes protocol.
+
+        See also: `AWS API Documentation <https://docs.aws.amazon.com/goto/WebAPI/glue-2017-03-31/StartJobRun>`_
+
+        Args:
+            job_name (str): The name of the job to use.
+            context (OpExecutionContext): The context of the currently executing Dagster op or asset.
+            extras (Optional[Dict[str, Any]]): Additional Dagster metadata to pass to the Glue job.
+            arguments (Optional[Dict[str, str]]): Arguments to pass to the Glue job Command
+            job_run_id (Optional[str]): The ID of the previous job run to retry.
+            allocated_capacity (Optional[int]): The amount of DPUs (Glue data processing units) to allocate to this job.
+            timeout (Optional[int]): The job run timeout in minutes.
+            max_capacity (Optional[float]): The maximum capacity for the Glue job in DPUs (Glue data processing units).
+            security_configuration (Optional[str]): The name of the Security Configuration to be used with this job run.
+            notification_property (Optional[Mapping[str, Any]]): Specifies configuration properties of a job run notification.
+            worker_type (Optional[str]): The type of predefined worker that is allocated when a job runs.
+            number_of_workers (Optional[int]): The number of workers that are allocated when a job runs.
+            execution_class (Optional[Literal["FLEX", "STANDARD"]]): The execution property of a job run.
+        """
+        with open_pipes_session(
+            context=context,
+            message_reader=self._message_reader,
+            context_injector=self._context_injector,
+            extras=extras,
+        ) as session:
+            arguments = arguments or {}
+            assert arguments is not None
```

Comment:
We try to avoid `assert` outside of tests. Occasionally it's needed for type-checker-legible runtime checks, but not in this case. The assertion here is already guaranteed by the above line.